pilot/pkg/bootstrap/server.go

```diff
@@ -493,22 +491,18 @@ func (s *Server) initSDSServer() {
 	if s.kubeClient == nil {
 		return
 	}
-	if !features.EnableXDSIdentityCheck {
-		// Make sure we have security
-		log.Warnf("skipping Kubernetes credential reader; PILOT_ENABLE_XDS_IDENTITY_CHECK must be set to true for this feature.")
-	} else {
-		creds := kubecredentials.NewMulticluster(s.clusterID)
-		creds.AddSecretHandler(func(name string, namespace string) {
-			s.XDSServer.ConfigUpdate(&model.PushRequest{
-				Full:           false,
-				ConfigsUpdated: sets.New(model.ConfigKey{Kind: kind.Secret, Name: name, Namespace: namespace}),
 
-				Reason: []model.TriggerReason{model.SecretTrigger},
-			})
+	creds := kubecredentials.NewMulticluster(s.clusterID)
+	creds.AddSecretHandler(func(name string, namespace string) {
```

Comment:
First time I see this code - could you add a small comment on what it does ? Why would we do a push on secrets only of XDS identity check is enabled ?